File: benchmarks/python/lcg.py
import os

def lcg(n: int) -> int:
    a = 1664525
    c = 1013904223
    mask = 0xFFFFFFFF  # emulate 32-bit wrap-around
    x = 0
    for _ in range(n):
        x = (a * x + c) & mask
    return x

N = int(os.environ.get("N", "50000000"))
print(lcg(N))

File: benchmarks/ruby/lcg.rb
def lcg(n)
  a = 1664525
  c = 1013904223
  mask = 0xFFFFFFFF
  x = 0
  n.times do
    x = (a * x + c) & mask
  end
  x
end

n = (ENV['N'] || '50000000').to_i
puts lcg(n)
